azure/services/subnets/spec.go

```diff
@@ -54,15 +56,90 @@ func (s *SubnetSpec) OwnerResourceName() string {
 	return s.VNetName
 }
 
+// verifyRouteTableImmutable validates that an immutable change isn't made to the subnet's route table.
+func (s *SubnetSpec) verifyRouteTableImmutable(existingSubnet network.Subnet) error {
```

Comment:
I'm not happy with these function names. If you have a better suggestion I'm happy to implement them. It's basically meant to validate that the changes are not made to those fields since they are immutable. I originally had them in the `Parameters` function but it was failing the cyclomatic complexity checks.

Reply:
why is this validation happening here and not in the webhook?

Reply:
Sounds like you just answered https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2635#issuecomment-1261222657 for me. :)

Reply:
Yes, thanks Cecile, sorry I didn't catch this earlier. We want these validations to fire before the resource updates ever enter into the reconciliation.

Reply:
This has been updated